### DISCUSSION TITLE
[Bug] Error API Endpoint usage

### DISCUSSION THREAD

**Discussion by Lostsite** [lobehub/lobehub#2171] [General Question]
### 💻 系统环境

Windows

### 📦 部署环境

Other

### 🌐 浏览器

Firefox

### 🐛 问题描述

I have 2 api endpoints set in ".env" 
```
OPENAI_API_KEY=sk-xxxx

OPENROUTER_API_KEY=sk-xxxx
```

and the model list
```
OPENROUTER_MODEL_LIST=-all,+mistralai/mixtral-8x22b-instruct,

OPENAI_MODEL_LIST=gpt-4-turbo-2024-04-09
```

However, when I choose "gpt-4-turbo-2024-04-09" as my model number in chart, it actually use the `OPENROUTER_API_KEY` (I can see usage on the openrouter, not the `OPENAI_API_KEY` (No usage from openai).

### 🚦 期望结果

Based on the model list I provide (`OPENROUTER_MODEL_LIST` vs `OPENAI_MODEL_LIST=`), use different API key and end point. (If both model list have same model, choose API endpoint and key from the selection (In the chart, I can select model from OpenAI or OpenRouter))

### 📷 复现步骤

git clone
Edit `.env`
Set `OPENAI_API_KEY=sk-xxxx` and `OPENROUTER_API_KEY`
Set `OPENROUTER_MODEL_LIST` and `OPENAI_MODEL_LIST=`

Set `OPENAI_PROXY_URL` to OpenRouter's API endpoint (I did this before OpenRouter become part of official support)
Call the "GPT-4" from Openai (doesn't matter, since `OPENAI_PROXY_URL` been set to OpenRouter, it will go to OpenRouter anyway).
Remove `OPENAI_PROXY_URL` froom `.env`
Call the "GPT-4" from Openai once again --- This request went to OpenRouter, not OpenAI

### 📝 补充信息

I use "Local Development" as my usage case (git clone + $ pnpm install + $ pnpm dev) on windows.

**Comment by Lostsite** [comment]
I have figure out the problem....

It won't read the api key and endpoint from ".env" file, It needs to be maunally set inside the "Language Model" section. I changed the ".env" settings. But been overwritten by "Language Model" section.

Expect:
Correct read the api key setting in the ".env", and use ".env" to overwrite other settings.

**Comment by arvinxx** [comment]
you can reset your local setting config in the setting -> common page.  the env override order is :

- firstly `.env`
- secondly user config
